src/main/java/org/swisspush/redisques/performance/UpperBoundParallel.java

```diff
@@ -0,0 +1,200 @@
+package org.swisspush.redisques.performance;
+
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.BiConsumer;
+
+import static java.lang.Thread.currentThread;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Level 1: KISS says do everything sequentially. Stop here! You're
```

Comment:
Propose to say in the first sentence what it is, and then explain in detail. Now it seems the other way round.

Reply:
Oh yes :) Thx.